src/Core/Parser/SyntaxTree/Expression/CastingExpressionNode.php

```diff
@@ -113,8 +115,16 @@ protected static function convertToArray($variable)
         if (is_array($variable)) {
             return $variable;
         }
-        if (is_string($variable) && strpos($variable, ',')) {
-            return array_map('trim', explode(',', $variable));
+        if (is_string($variable)) {
+            if (preg_match('/^[\[\{].*[\]\}]$/', $variable)) {
```

Comment:
I guess a `trim()` should be used here. The regexp would not match if there is one or more spaces at the beginning or the end. `json_decode()` properly handles this.

```
$myjsonstring = ' {"k1": "v1"} ';
```

would not be matched. See: https://3v4l.org/XWMi2

Reply:
Thanks. You should also fix the cgl issues, see PHP 8.1 checks.

Reply:
Just done that ;-)

Reply:
Don't want to bother you, but maybe we can add at least one test for the space/trim addition ?

Reply:
Sure, done.